### PR TITLE
Make site metadata fetch endpoint require auth.

### PR DESCRIPTION
**Pull request by dessalines**
@Nothing4You pointed out, that the only place we really need to do site metadata fetching externally, is from the `create_post` page. So it should also probably require auth.

I tested this also and it works.

### CODE REVIEW

File: crates/api/src/post/get_link_metadata.rs [hunk 1]
```diff
@@ -14,6 +15,7 @@ use url::Url;
 pub async fn get_link_metadata(
   data: Query<GetSiteMetadata>,
   context: Data<LemmyContext>,
+  _local_user_view: LocalUserView,
```
Comment (SleeplessOne1917) [inline crates/api/src/post/get_link_metadata.rs:18]: Is the unused param what handles adding the auth requirement?

Comment (dessalines) [reply crates/api/src/post/get_link_metadata.rs:18]: Yep. The auth middleware will throw an `invalid_auth` error if its not an `Option<LocalUserView>`

Comment (Nothing4You) [reply crates/api/src/post/get_link_metadata.rs:18]: here's the logic for that: https://github.com/LemmyNet/lemmy/blob/128e78f7c22279fec27d3213a05b440cb1f8db59/crates/db_views/src/local_user_view.rs#L128-L138

**Review by SleeplessOne1917 — APPROVED** [review]

File: crates/api/src/post/get_link_metadata.rs (hunk 1 above)
Comment (dullbananas) [inline crates/api/src/post/get_link_metadata.rs:18]:
```suggestion
  // Require an account for this API
  _local_user_view: LocalUserView,
```

**Pull request merged**